### DISCUSSION TITLE
Problems about Simulink and Yalmip

### DISCUSSION THREAD

**Discussion by JasonLiu555** [yalmip/YALMIP#979] [General discussion] [answered]
Dear Jahon,

I worte a S-function to use co-simulation of Carsim and Matlab/Simulink so that achieving the Intelligent vehicle trajectory tracking control. However, I found the simulations run slowly. Could you  give some suggestions?

Thanks in advance!

**Comment by johanlofberg** [comment, marked as answer]
If you are using the optimizer approach as shown in the example, and you are using a good solver, there is nothing more you can do from a YALMIP perspective.

**Reply by JasonLiu555** [reply]
I used solver penbmi and sedumi to solve Linear matrix inequality（LMI）, the 20s simulation actually took me 6 minutes，so there is nothing more can I Speed up simulation？

**Comment by johanlofberg** [comment]
penbmi is something I would never use. It is very unstable and not updated for many years. If you are solving LMIs repeatedly, you should use an optimizer setup with Mosek as the solver.Sedumi is ok-ish, but typically less stable and slower than mosek

**Reply by JasonLiu555** [reply]
Thanks, I  used penbmi for solving a bilinear matrix inequality（BMI）, I will try mosek next time.

**Reply by johanlofberg** [reply]
Solving BMIs repeatedly as part of a controller is something I would expect to never work. That fact that PENBMI doesn't crash and that it even returns something sensible is mostly luck.
